moveit_ros/planning/planning_request_adapter_plugins/src/check_start_state_bounds.cpp

```diff
@@ -144,33 +144,36 @@ class CheckStartStateBounds : public planning_interface::PlanningRequestAdapter
         }
         default:
         {
-          if (!start_state.satisfiesBounds(jmodel))
-          {
-            std::stringstream joint_values;
-            std::stringstream joint_bounds_low;
-            std::stringstream joint_bounds_hi;
-            const double* p = start_state.getJointPositions(jmodel);
-            for (std::size_t k = 0; k < jmodel->getVariableCount(); ++k)
-            {
-              joint_values << p[k] << ' ';
-            }
-            const moveit::core::JointModel::Bounds& b = jmodel->getVariableBounds();
-            for (const moveit::core::VariableBounds& variable_bounds : b)
-            {
-              joint_bounds_low << variable_bounds.min_position_ << ' ';
-              joint_bounds_hi << variable_bounds.max_position_ << ' ';
-            }
-            RCLCPP_ERROR(logger_,
-                         "Joint '%s' from the starting state is outside bounds by: [%s] should be in "
-                         "the range [%s], [%s].",
-                         jmodel->getName().c_str(), joint_values.str().c_str(), joint_bounds_low.str().c_str(),
-                         joint_bounds_hi.str().c_str());
-          }
+          break;  // do nothing
+        }
+      }
+
+      // Check the joint against its bounds.
+      if (!start_state.satisfiesBounds(jmodel))
```

Comment:
Should this condition set `changed_req` to `true` and actually update the start_state variable to be within bounds using `enforceBounds()`?

The old code had this so long as the state was within some specified parameter threshold, which has since been removed.

Reply:
There was a discussion in https://github.com/moveit/moveit2/issues/2849

@sjahr stated "In main I replaced the FixStartState adapter in https://github.com/moveit/moveit2/pull/2429 with a simple [CheckStartStateCollision](https://github.com/moveit/moveit2/blob/main/moveit_ros/planning/planning_request_adapter_plugins/src/check_start_state_collision.cpp) adapter. My rationale is that an adapter that introduces small jumps is un-intuitive and might lead to undesired behavior, especially given that is was used almost everywhere as a default. I think the decision about what to do when the start state is in collision should be made in a higher level logic than the planning pipeline."

Reply:
I'd still state that, what do you think?

Reply:
This is fine!